run_matrix.py

```diff
@@ -375,12 +375,15 @@ def run_interface_tests(
         )
         test_results[interface] = results_per_version
 
-        # Running in GitHub actions with the maintainer set on the test.
-        if os.getenv("GITHUB_ACTIONS"):
-            for version, tests_per_role in version_to_roles.items():
-                maintainer = tests_per_role.get("maintainer")
-                if maintainer and test_failed(results_per_version[version]):
-                    failed = True
+        for version, tests_per_role in version_to_roles.items():
```

Comment:
Took me a while to find this.

https://github.com/canonical/pytest-interface-tester/blob/fddd30b37022bada9c0bf83877bd5dbdbeed8521/interface_tester/collector.py#L316

I wonder if a local comment or some fancy local type using `Annotated` could be used to make the test collection result/bundle format more understandable.